mapie/time_series_regression.py

```diff
@@ -0,0 +1,325 @@
+        self.conformity_scores_ = np.asarray(y) - y_pred
+        return self
+
+    def partial_fit(
+        self,
+        X: ArrayLike,
+        y: ArrayLike,
+    ) -> MapieTimeSeriesRegressor:
+        """
+        Update the ``conformity_scores_`` attribute when data with known labels
+        are available.
+
+        Parameters
+        ----------
+        X : ArrayLike of shape (n_samples, n_features)
+            Input data.
+
+        y : ArrayLike of shape (n_samples,)
+            Input labels.
+
+        Returns
+        -------
+        MapieTimeSeriesRegressor
+            The model itself.
+        """
+        y_pred, _ = self.predict(X, alpha=0.5, ensemble=True)
+        new_conformity_scores_ = np.asarray(y) - np.asarray(y_pred)
+        new_conformity_scores_ = new_conformity_scores_[
+            ~np.isnan(new_conformity_scores_)
+        ]
+
+        cut_index = min(
+            len(new_conformity_scores_), len(self.conformity_scores_)
+        )
+        self.conformity_scores_ = np.concatenate(
+            [
+                self.conformity_scores_[cut_index:],
+                new_conformity_scores_,
+            ],
+            axis=0,
+        )
+        return self
+
+    def predict(
+        self,
+        X: ArrayLike,
+        ensemble: bool = False,
+        alpha: Optional[Union[float, Iterable[float]]] = None,
+        JAB_Like=False,
+    ) -> Union[NDArray, Tuple[NDArray, NDArray]]:
+        """
+        ``predict`` method correspond to the ``MapieRegressor``'s one with the
+        argument ``JAB_Like`` in more. In case  ``JAB_Like`` is ``False``,
+        predictions correspond to [6]. In case it is True, is is another
+        implementation inspired of [2] with wider PIs. The second method is
+        much slower because of PIwise optimization.
+
+        Parameters
+        ----------
+        JAB_Like : boolean, default False
+            Whether to use the implementation of [6] or an implementation
+            closer to [2].
+        """
+
+        # Checks
+        check_is_fitted(self, self.fit_attributes)
+        self._check_ensemble(ensemble)
+        alpha = cast(Optional[NDArray], check_alpha(alpha))
+        X = check_array(X, force_all_finite=False, dtype=["float64", "object"])
+        y_pred = self.single_estimator_.predict(X)
+        n = len(self.conformity_scores_)
+
+        if alpha is None:
+            return np.array(y_pred)
+        else:
+            alpha_np = cast(NDArray, alpha)
+            check_alpha_and_n_samples(alpha_np, n)
+
+            if (
+                (not JAB_Like)
+                or (self.method in ["naive", "base"])
+                or (self.cv == "prefit")
+            ):
+                # This version of predict corresponds to [6].
+                # Its PIs are closed to the oracle's ones.
+                betas_0 = np.full_like(alpha_np, np.nan, dtype=float)
+                for ind, _alpha in enumerate(alpha_np):
```

Comment:
Add a comment describing what is done in the loop "# optimize the beta parameter to ..."

Reply:
Or even place it in a dedicated method of `TimeSeriesRegressor` ?

Reply:
Done the second option